### PR TITLE
ar71xx: fix TP-Link Archer C7 v5 switch LEDs

### PR DESCRIPTION
**Pull request by Rinki13**
Signed-off-by: Rinki13 <rinki13@gmail.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

### CODE REVIEW

File: target/linux/ar71xx/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -155,6 +155,16 @@ archer-c7-v4)
 	ucidef_set_led_usbdev "usb1" "USB1" "$board:green:usb1" "1-1"
 	ucidef_set_led_usbdev "usb2" "USB2" "$board:green:usb2" "2-1"
 	;;
+archer-c7-v5)
```
Comment (aep) [inline target/linux/ar71xx/base-files/etc/board.d/01_leds]: since its almost identical to the archer-c7-v4 , can you merge them?

**Review by aep — CHANGES_REQUESTED** [review]

File: target/linux/ar71xx/base-files/etc/board.d/01_leds (hunk 1 above)
Comment (Rinki13) [reply target/linux/ar71xx/base-files/etc/board.d/01_leds]: but archer-c7-v5 doesn't have usb2.

Comment (aep) [reply target/linux/ar71xx/base-files/etc/board.d/01_leds]:
```
case "$board" in 
    archer-c7-v4)
       ucidef_set_led_usbdev "usb2" "USB2" "$board:green:usb2" "2-1"
```

Comment (Rinki13) [reply target/linux/ar71xx/base-files/etc/board.d/01_leds]: done

Comment (Rinki13) [reply target/linux/ar71xx/base-files/etc/board.d/01_leds]: Done

**Review by Rinki13 — COMMENTED** [review]
requested changes are done

**Comment by aep** [comment]
thanks. looks good to me. (i'm not an openwrt maintainer, no idea when they will look at it)

**Comment by Rinki13** [comment]
@aep : can you approve pull request.

**Review by aep — APPROVED** [review]

**Comment by rotanid** [comment]
i also tested this and the switch LEDs work like they should

**Comment by rotanid** [comment]
(i had to delete my first comment).
@Rinki13 what do you mean by this PR "fix switch LEDs" ? what was working and what was not, and what should the behaviour with this patch be?

**Comment by Rinki13** [comment]
@rotanid : without patch only power led was green and rest of the leds off. after applying patch connected interface leds became green.

**Comment by rotanid** [comment]
@Rinki13 ok, but then the activity is still missing (blinking if traffic occurs)
maybe @blocktrron can give you a hint on how to fix this, too

**Comment by JamesT42** [comment]
This works after patching into 18.06 codebase. But it fixes all LEDs, not just switch LEDs.

**Comment by rotanid** [comment]
ack, works

**Comment by JamesT42** [comment]
I have to correct myself: USB does not seem to work by default, one has to switch the LED from ...wlan2g to ...USB and setup usbdev as trigger by hand.

**Comment by rotanid** [comment]
@JamesT42 is this a suggested change to this PR?

**Comment by JamesT42** [comment]
> 
> 
> @JamesT42 is this a suggested change to this PR?

Yes, but I do not have the time right now to find the problem myself. Maybe the change by aep is already enough?

**Comment by rotanid** [comment]
@JamesT42 huh? i thought your comment was about this PR not about openwrt without this PR? if the latter is the case, your comment is rather contraproductive...

**Comment by JamesT42** [comment]
> @JamesT42 huh? i thought your comment was about this PR not about openwrt without this PR? if the latter is the case, your comment is rather contraproductive...

Of course my comment is about this PR. See https://github.com/openwrt/openwrt/pull/1483 for possible fix for my USB LED problem. But I can not test it anymore, because I do not have the device any more.

**Comment by rotanid** [comment]
ah right. @Rinki13 could you please include the fixes from #1483 in your PR?

**Comment by Rinki13** [comment]
sure. i will make it.

**Comment by Rinki13** [comment]
@rotanid : I have included #1483 changes in my pull request.

**Comment by NeoRaider** [comment]
@Rinki13 Thanks, the patch itself looks good to me. As described in https://openwrt.org/submitting-patches , commit author and signed-off line must contain your real name, not your Github username.

**Comment by Rinki13** [comment]
added real name in signoff

**Comment by gostek-1** [comment]
Hi there, I've had not working LED problem on my C7 v5, i did workaround and achived working LEDs on SNAPSHOT r8422-c5bf408  git-18.309.37192-a6d96e0 build and /etc/config/system config below:
`config led
	option default '0'
	option name 'wlan2g'
	option sysfs 'archer-c7-v5:green:wlan2g'
	option trigger 'phy1tpt'

config led
	option default '0'
	option name 'wlan5g'
	option sysfs 'archer-c7-v5:green:wlan5g'
	option trigger 'phy0tpt'

config led
	option default '0'
	option name 'wan'
	option sysfs 'archer-c7-v5:green:wan'
	option trigger 'switch0'
	option port_mask '0x2'

config led
	option default '0'
	option name 'lan1'
	option sysfs 'archer-c7-v5:green:lan1'
	option trigger 'switch0'
	option port_mask '0x6'

config led
	option default '0'
	option name 'lan2'
	option sysfs 'archer-c7-v5:green:lan2'
	option trigger 'switch0'
	option port_mask '0x8'

config led
	option default '0'
	option name 'lan3'
	option sysfs 'archer-c7-v5:green:lan3'
	option trigger 'switch0'
	option port_mask '0x10'

config led
	option default '0'
	option name 'lan4'
	option sysfs 'archer-c7-v5:green:lan4'
	option trigger 'switch0'
	option port_mask '0x20'`

**Comment by neveripe** [comment]
> config led
> option default '0'
> option name 'lan1'
> option sysfs 'archer-c7-v5:green:lan1'
> option trigger 'switch0'
> option port_mask '0x6'

Should be 0x4, not 0x6.

I'm also using this workaround, but I'm not able to make them blink when transfer is happening. Wonder if this is possible to fix or need a new build with proper patch?

**Comment by blocktrron** [comment]
@neveripe Switchports only blink on data activity when they are controlled by the switch chip itself. THis is not the case here, so they don't blink.

**Comment by NeoRaider** [comment]
Thanks, merged!

I noticed that the commit author name was still your Github username, I fixed that up when merging.

**Comment by neveripe** [comment]
> @neveripe Switchports only blink on data activity when they are controlled by the switch chip itself. THis is not the case here, so they don't blink.

Is there a way to make switch chip to control those?

**Comment by JamesT42** [comment]
I can not reproduce your problem, my leds on a V5 correspond to the correct ports left/right/whatever if you look from the front.

**Comment by gostek-1** [comment]
> I can not reproduce your problem, my leds on a V5 correspond to the correct ports left/right/whatever if you look from the front.

Do you have EU version? mine was.
And pls tell mi why oh why RP-SMA is gone ;(

**Comment by TZanke** [comment]
> I can not reproduce your problem, my leds on a V5 correspond to the correct ports left/right/whatever if you look from the front.

I bought the device from amazon Germany, it has a German written EU-Sticker at the package backside and a Spanish written "Computer Hoy" sticker at the front. 

Device itself is labeled as Archer C7 V5 (EU)

> And pls tell mi why oh why RP-SMA is gone ;(

It was amazingly cheap, so i think V5 is a new cost efficient version. =)

**Comment by gostek-1** [comment]
ech, bu why oh why I thought that C1200 was cheap home version AC1750 was always a little level higher home device, not premium but always 2USB and 802.11 AC.

**Comment by gostek-1** [comment]
So the current snapshot has merged LED fix?

**Comment by rotanid** [comment]
yes

**Pull request closed without merging**